android/java/org/chromium/chrome/browser/preferences/BraveSearchEngineUtils.java

```diff
@@ -0,0 +1,83 @@
+    private static final String STANDARD_DSE_KEYWORD = "standard_dse_keyword";
+    private static final String PRIVATE_DSE_SHORTNAME = "private_dse_shortname";
+    private static final String STANDARD_DSE_SHORTNAME = "standard_dse_shortname";
+
+    static public String getDSEShortName(boolean isPrivate) {
+        String defaultSearchEngineName = null;
+        TemplateUrl dseTemplateUrl =
+                TemplateUrlServiceFactory.get().getDefaultSearchEngineTemplateUrl();
+        if (dseTemplateUrl != null)
+            defaultSearchEngineName = dseTemplateUrl.getShortName();
+
+        return ContextUtils.getAppSharedPreferences().getString(
+            isPrivate ? PRIVATE_DSE_SHORTNAME : STANDARD_DSE_SHORTNAME, defaultSearchEngineName);
+    }
+
+    static public String getDSEKeyword(boolean isPrivate) {
+        String defaultSearchEngineName = null;
+        TemplateUrl dseTemplateUrl =
+                TemplateUrlServiceFactory.get().getDefaultSearchEngineTemplateUrl();
+        if (dseTemplateUrl != null)
+            defaultSearchEngineName = dseTemplateUrl.getKeyword();
+
+        return ContextUtils.getAppSharedPreferences().getString(
+            isPrivate ? PRIVATE_DSE_KEYWORD : STANDARD_DSE_KEYWORD, defaultSearchEngineName);
+    }
+
+    static public void updateActiveDSEWith(boolean isPrivate) {
+        TemplateUrlServiceFactory.get().setSearchEngine(getDSEKeyword(isPrivate));
+    }
+
+    static public void setDSEPrefs(TemplateUrl templateUrl, boolean isPrivate) {
+        SharedPreferences.Editor sharedPreferencesEditor =
+            ContextUtils.getAppSharedPreferences().edit();
+        sharedPreferencesEditor.putString(
+            isPrivate ? PRIVATE_DSE_KEYWORD : STANDARD_DSE_KEYWORD, templateUrl.getKeyword());
+        sharedPreferencesEditor.putString(
+            isPrivate ? PRIVATE_DSE_SHORTNAME : STANDARD_DSE_SHORTNAME, templateUrl.getShortName());
+        sharedPreferencesEditor.apply();
+    }
+
+    static public void initializeBraveSearchEngineStates(TabModelSelector tabModelSelector) {
+        tabModelSelector.addObserver(new SearchEngineTabModelSelectorObserver(tabModelSelector));
+
+        initializeDSEPrefs();
+        // Initially set standard dse as a active DSE.
+        updateActiveDSEWith(false);
+    }
+
+    static private void initializeDSEPrefs() {
+        // At first run, we should set initial default prefs to each standard/private DSE prefs.
+        // Those pref values will be used until user change DES options explicitly.
+        final String notInitialized = "notInitialized";
+        if (notInitialized.equals(ContextUtils.getAppSharedPreferences().getString(STANDARD_DSE_SHORTNAME, notInitialized))) {
+            TemplateUrl templateUrl =
+                TemplateUrlServiceFactory.get().getDefaultSearchEngineTemplateUrl();
+
+            SharedPreferences.Editor sharedPreferencesEditor =
+            ContextUtils.getAppSharedPreferences().edit();
```

Comment:
Nit: The indentation here seems a little confusing, should it be a little further in since it continues the previous line?

Reply:
Right. wrong indentation.